### PR TITLE
fix: add enable manual linking option to auth config

### PR DESCRIPTION
**Pull request by mosnicholas**
## What kind of change does this PR introduce?

Allow us to test manual linking in localhost

## What is the current behavior?

[Please link any relevant issues here.](https://github.com/supabase/gotrue/pull/1317#issuecomment-1892937708)

## What is the new behavior?

It allows you to specify config.toml value that lets us test manual linking locally

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by mosnicholas** [comment]
@hf / @sweatybridge / @kangmingtay I can't request review, so pinging here for visibility :)

File: internal/start/start.go [hunk 1]
```diff
@@ -380,6 +380,7 @@ EOF
 			"GOTRUE_SITE_URL=" + utils.Config.Auth.SiteUrl,
 			"GOTRUE_URI_ALLOW_LIST=" + strings.Join(utils.Config.Auth.AdditionalRedirectUrls, ","),
 			fmt.Sprintf("GOTRUE_DISABLE_SIGNUP=%v", !utils.Config.Auth.EnableSignup),
+			fmt.Sprintf("GOTRUE_SECURITY_MANUAL_LINKING_ENABLED=%v", utils.Config.Auth.EnableManualLinking),
```
Comment (sweatybridge) [inline internal/start/start.go:383]: Could you move this to line 418 so it's grouped with other GOTRUE_SECURITY_ prefixed env vars?

File: internal/utils/config.go [hunk 2]
```diff
@@ -310,10 +310,11 @@ type (
 		EnableRefreshTokenRotation bool `toml:"enable_refresh_token_rotation"`
 		RefreshTokenReuseInterval  uint `toml:"refresh_token_reuse_interval"`
 
-		EnableSignup bool  `toml:"enable_signup"`
-		Email        email `toml:"email"`
-		Sms          sms   `toml:"sms"`
-		External     map[string]provider
+		EnableSignup        bool  `toml:"enable_signup"`
+		EnableManualLinking bool  `toml:"enable_manual_linking"`
```
Comment (sweatybridge) [inline internal/utils/config.go:314]: Ditto. Let's move this below line 311

File: internal/utils/templates/init_config.toml [hunk 3]
```diff
@@ -81,6 +81,8 @@ enable_refresh_token_rotation = true
 refresh_token_reuse_interval = 10
 # Allow/disallow new user signups to your project.
 enable_signup = true
+# Allow/disallow testing manual linking of accounts
+enable_manual_linking = false
```
Comment (sweatybridge) [inline internal/utils/templates/init_config.toml:85]: Remember to also add `enable_manual_linking = true` to init_config.test.toml

**Review by sweatybridge — COMMENTED** [review]
Thanks for the contribution. Just a few nitpicks.

**Comment by mosnicholas** [comment]
Thanks @sweatybridge, updated :)

File: internal/utils/config.go [hunk 4]
```diff
@@ -734,7 +735,7 @@ func InitConfig(params InitParams, fsys afero.Fs) error {
 	if err := MkdirIfNotExistFS(fsys, filepath.Dir(ConfigPath)); err != nil {
 		return err
 	}
-	f, err := fsys.OpenFile(ConfigPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0644)
+	f, err := fsys.OpenFile(ConfigPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0o644)
```
Comment (sweatybridge) [inline internal/utils/config.go:738]:
```suggestion
	f, err := fsys.OpenFile(ConfigPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0644)
```

Is this change intentional? I would suggest reverting since it's unrelated

**Review by sweatybridge — APPROVED** [review]

File: internal/utils/config.go (hunk 4 above)
Comment (mosnicholas) [reply internal/utils/config.go:738]: Nope, thank you. I used `gofumpt` which did this, I missed that change.

**Comment by mosnicholas** [comment]
@sweatybridge thanks for the guiding! Is it easy to get this out? :D would love to start building and testing against it.

**Comment by sweatybridge** [comment]
Yup, it's good to go. Once main branch is done building, the updated cli will be released to beta channel. 

You can run it with `npx supabase@beta help`
